python/cudf/cudf/core/column/categorical.py

```diff
@@ -1046,11 +1071,27 @@ def __cuda_array_interface__(self) -> Mapping[str, Any]:
     def to_pandas(
         self, index: ColumnLike = None, nullable: bool = False, **kwargs
     ) -> pd.Series:
-        signed_dtype = min_signed_type(len(self.categories))
-        codes = self.cat().codes.astype(signed_dtype).fillna(-1).to_array()
-        categories = self.categories.to_pandas()
+
+        if self.categories.dtype.kind == "f":
+            new_mask = bools_to_mask(self.notnull())
+            col = column.build_categorical_column(
+                categories=self.dtype.categories._values,
+                codes=column.as_column(
+                    self.codes.base_data, dtype=self.codes.dtype
+                ),
+                mask=new_mask,
```

Comment:
I'm not sure if `new_mask` would be offset aware here. It's generated from the result of `notnull` which would return a new `cudf::column` under the hood.

I think this is a situation we need to use `self.codes` and pass 0 as the offset and `self.codes.size` as the size.

Reply:
Done 👍 Along with this changed I fixed a slight issue in repr too.